amti/clis/preview.py

```diff
@@ -0,0 +1,100 @@
+"""CLI for running a simple web server to preview HIT"""
+
+import os
+import json
+from http.server import BaseHTTPRequestHandler, HTTPServer
+from urllib.parse import urlparse
+
+import click
+import jinja2
+
+from amti import actions
+from amti import settings
+from amti import utils
+
+
+class MyServer(HTTPServer):
+    def __init__(
+        self,
+        server_address,
+        RequestHandlerClass,
+        template_path=None,
+        data_path=None,
+    ):
+        super().__init__(server_address, RequestHandlerClass)
+        self.template_path = template_path
+        self.data_path = data_path
+
+
+class MyHandler(BaseHTTPRequestHandler):
+    def _set_headers(self):
+        self.send_response(200)
+        self.send_header('Content-type', 'text/html')
+        self.end_headers()
+
+    def do_GET(self): # handle GET request
+        self._set_headers()
+
+        # Get data index from query (optional)
+        query = urlparse(self.path).query
+        try:
+            query_components = dict(qc.split("=") for qc in query.split("&"))
+        except ValueError:
+            query_components = {}
+        index = int(query_components.get('id', '0'))
+
+        # Render
+        with open(self.server.template_path, 'r') as template_file:
+            question_template = jinja2.Template(template_file.read())
+        with open(self.server.data_path, 'r') as data_file:
+            ln_data = {}
+            for i, f in enumerate(data_file):
+                if i == index:
+                    ln_data = json.loads(f)
+                    break
+        self.wfile.write(question_template.render(**ln_data).encode())
+
+def run(
+    template_path,
+    data_path,
+    port,
+    server_class=MyServer,
+    handler_class=MyHandler,
+):
+    server_address = ('', port)
+    httpd = server_class(server_address, handler_class, template_path=template_path, data_path=data_path)
+    print(f"Starting server on port {port}...")
+    httpd.serve_forever()
```

Comment:
I think we can place this logic in the `preview_batch` function for now. The change would make the file a little smaller / simpler, and we can always pull it out later if we find a use case.

Reply:
Rather than a query parameter, we can probably use a regex: `re.compile(r"/hits/(\d+)")` to pull the parameter straight from the URL. The advantage would be that it makes the HIT number "required" and then we can get rid of the default HIT number (simplifying the server's behavior and logic).